library/src/juke.cpp

```diff
@@ -0,0 +1,60 @@
+
+#include <capo/engine.hpp>
+#include <juke/core/AudioFile.hpp>
+#include <juke/core/MediaError.hpp>
+#include <juke/core/XMFile.hpp>
+#include <juke/juke.hpp>
+#include <print>
+
+namespace juke {
+
+using namespace std::chrono_literals;
+
+Jukebox::Jukebox(capo::IEngine& audio_engine) : m_source(audio_engine.create_source()) {
+	if (!m_source) { throw std::runtime_error{"Failed to create Audio Source"}; }
+}
+
+bool Jukebox::load_media(std::filesystem::path const& path) {
+	try {
+		auto media_file = std::unique_ptr<IMediaFile>{};
+		if (XMFile::is_xm(path)) {
+			media_file = std::make_unique<XMFile>(path);
+		} else {
+			media_file = std::make_unique<AudioFile>(path);
+		}
+		m_source->stop();
+		if (!media_file->bind_to(*m_source)) {
+			std::println("ERROR: Failed to bind Source to PCM");
+			return false;
+		}
+		m_media_file = std::move(media_file);
+	} catch (MediaError const& e) { // only MediaErrors are caught here, others are raised all the way to main.
+		std::println("ERROR: {}", e.what());
+		return false;
+	}
+
+	return true;
+}
+
+void Jukebox::play(bool looping) {
+	m_source->play();
+	m_status = m_source->is_playing() ? MediaStatus::playing : MediaStatus::stopped;
+	m_source->set_looping(looping);
```

Comment:
1. Call `set_looping()` before calling `play()`, just in case it finishes playing before you manage to set it to loop 😛 
2. IMO `m_status` is unnecessary, and risks going out of sync: it may be playing now but could have stopped in the next second